app/gui/view/graph-editor/src/lib.rs

```diff
@@ -1781,17 +1799,27 @@ impl GraphEditorModel {
             add_node_button,
             styles_frp,
             selection_controller,
+            execution_mode_selector,
         }
         .init()
     }
 
     fn init(self) -> Self {
-        self.add_child(&self.breadcrumbs);
         let x_offset = MACOS_TRAFFIC_LIGHTS_SIDE_OFFSET;
-        let y_offset = MACOS_TRAFFIC_LIGHTS_VERTICAL_CENTER + component::breadcrumbs::HEIGHT / 2.0;
+        let y_offset = MACOS_TRAFFIC_LIGHTS_VERTICAL_CENTER;
+        let traffic_light_width = traffic_lights_gap_width();
+
+        self.add_child(&self.execution_mode_selector);
+        self.execution_mode_selector
+            .set_x(traffic_light_width + execution_mode_selector::OVERALL_WIDTH / 2.0);
+        self.execution_mode_selector.set_y(y_offset + execution_mode_selector::HEIGHT / 2.0);
```

Comment:
The position of the execution mode selector should be reactive. It should probably be computed with FRP based on the width of traffic light. Even better, you can use our auto-layout to place execution mode next to traffic light. The description of how to use auto-layout is in `display::object` module.

Reply:
It actually is reactive based on the FRP updates below. This is just the default initialization. The reason I did not go with the layout is, that the traffic lights are: (a) not currently part of the Graph Editor and (b) as far as I am aware, are not always present, as on macOS there are the native controls placed there and this offset is applied without a corresponding EnsoGL component.

Reply:
(also some changes happening due to use of the theme here soon to be visible)

Reply:
Regarding auto-layouts - they work great when some items are not always present. Just FYI - you can create auto grid layout and when item is missing, other items will be layout correctly.